pkg/kademlia/routing.go

```diff
@@ -321,3 +322,77 @@ func (rt *RoutingTable) ConnSuccess(ctx context.Context, node *pb.Node) {
 		zap.L().Debug("connection success error:", zap.Error(err))
 	}
 }
+
+func splitBucket(bIDs []bucketID, bitDepth int) ([]bucketID, []bucketID) {
```

Comment:
Let's move these methods to a separate package, maybe `kademlia/routinggraph`? To avoid making the kademlia package larger.